src/broadcast.jl

```diff
@@ -19,7 +19,7 @@ _data(d::BroadcastedDictionary) = getfield(d, :data)
     sharetokens = _sharetokens(dicts...)
     I = keytype(dicts[1])
     Ts = Base.Broadcast.eltypes(data)
-    T = Core.Compiler.return_type(f, Ts)
+    T = Base.promote_op(f, Ts...)
```

Comment:
Maybe this should be `Ts.parameters...`?

Is that construction inferrable? (If not we can make a pure function with `@assume_effects`)

Reply:
so it turns out that there already is `Base.Broadcast.combine_eltypes` that does this functionality (except slightly more correctly).